### DISCUSSION TITLE
Making the CXF extension reactive

### DISCUSSION THREAD

**Discussion by Inithron** [quarkiverse/quarkus-cxf#1400] [Q&A] [answered]
Hi,
I have 2 Questions:
1. I have seen the [HTTP Async Transport](https://docs.quarkiverse.io/quarkus-cxf/dev/reference/extensions/quarkus-cxf-rt-transports-http-hc5.html) extension. Is something similar planned for the server side, to have a async/reactive CXF service?
2. The Quarkus REST client and services provide async/reactive support without an additional extension: https://quarkus.io/guides/rest#asyncreactive-support. Is it planned to do the same for the CXF clients and services?

**Comment by ppalaga** [comment, marked as answer]
> I have seen the [HTTP Async Transport](https://docs.quarkiverse.io/quarkus-cxf/dev/reference/extensions/quarkus-cxf-rt-transports-http-hc5.html) extension. Is something similar planned for the server side, to have a async/reactive CXF service?

Not at this time, unless there is enough demand from users or even better customers (if you are one, do not hesitate to reach out to Red Hat Support).

Even though there is no immediate plan to implement this, it would be interesting to see some basic service snippet showing what you actually expect. I wonder whether something like this can be reconciled with the existing JAX-WS and WS-* specs.

> The Quarkus REST client and services provide async/reactive support without an additional extension: https://quarkus.io/guides/rest#asyncreactive-support. Is it planned to do the same for the CXF clients and services?

No, there are no such plans at this point.
